src/main/java/pl/pandziak/ticketbookingapp/domain/reservation/ReservationValidator.java

```diff
@@ -0,0 +1,148 @@
+package pl.pandziak.ticketbookingapp.domain.reservation;
+
+import pl.pandziak.ticketbookingapp.domain.Room;
+import pl.pandziak.ticketbookingapp.domain.reservation.dtos.ReservationRequestDto;
+import pl.pandziak.ticketbookingapp.domain.reservation.dtos.TicketsDto;
+import pl.pandziak.ticketbookingapp.domain.screening.dtos.ScreeningResponseDto;
+import pl.pandziak.ticketbookingapp.domain.screening.dtos.SeatDto;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.springframework.util.Assert.isTrue;
+
+final class ReservationValidator {
+
+    private static final int MIN_USER_NAME_LEN = 3;
+    private static final String PERMITTED_CHARACTERS = "^[a-zA-Z]*[ ]*[-]?[ ]*[a-zA-Z]*$";
+
+    void validateInputFieldsNonNullOrEmpty(ReservationRequestDto reservationRequest) {
+        isTrue(reservationRequest.isValid(), "Every field should be filled");
+    }
+
+    void validateUserName(String name) {
+        isTrue(name.length() >= MIN_USER_NAME_LEN, "Name does not have the proper length of " + MIN_USER_NAME_LEN);
+        isTrue(isFirstLetterUpperCase(name), "Name does not start with lower case");
+        isTrue(name.matches(PERMITTED_CHARACTERS), "Illegal characters used in name");
+    }
+
+    void validateUserSurname(String surname) {
+        isTrue(surname.length() >= MIN_USER_NAME_LEN, "Surname does not have the proper length of " + MIN_USER_NAME_LEN);
+        isTrue(isFirstLetterUpperCase(surname), "Surname does not start with capital letter");
+        isTrue(twoPartsOfSurnameStartsWithUpperCase(surname), "Both parts of surname need to start with capital letter");
+        isTrue(surname.matches(PERMITTED_CHARACTERS), "Illegal characters used in surname");
+    }
+
+    void validateTickets(TicketsDto tickets) {
+        isTrue(!tickets.empty(), "Reservation should include at least one ticket");
+    }
+
+    void validateTime(LocalDateTime screeningTime) {
+        LocalDateTime allowedReservationTime = screeningTime.minusMinutes(15);
+        LocalDateTime now = LocalDateTime.now();
+
+        isTrue(now.isBefore(allowedReservationTime), "Reservations can be made only until 15 minutes before screening");
+    }
+
+    void validateChosenSeats(TicketsDto tickets, ScreeningResponseDto screeningResponse) {
+        validateTickets(tickets);
+
+        int maxColumn = Room.valueOf(screeningResponse.getScreeningDetails().getRoom()).getColumns();
```

Comment:
Dlaczego max?

Reply:
Jest to zmienna określająca numer ostatniego miejsca w rzędzie. Może zamiast 'max', 'last' byłoby bardziej zrozumiałe. Ostatni rząd wykorzystuję w późniejszych walidacjach (m.in. żeby uniknąć IndexOutOfBoundException).